packages/devtools_shared/lib/src/server/server_api.dart

```diff
@@ -68,23 +80,35 @@ class ServerApi {
         return api.getCompleted(json.encode(true));
       case apiGetDevToolsFirstRun:
         // Has DevTools been run first time? To bring up analytics dialog.
+        final isFirstRun =
+            _devToolsUsage.isFirstRun || analytics.shouldShowMessage;
         return api.getCompleted(
-          json.encode(_devToolsUsage.isFirstRun),
+          json.encode(isFirstRun),
         );
       case apiGetDevToolsEnabled:
         // Is DevTools Analytics collection enabled?
+        final isEnabled =
+            _devToolsUsage.analyticsEnabled && analytics.telemetryEnabled;
```

Comment:
will this mess up existing DevTools users who are opted into analytics for their first run? IIRC `analytics.telemetryEnabled` won't be true for users on their first run no matter what their opt in state is - is that correct?

Reply:
So there a few different scenarios and if the user has already interacted with `package:unified_analytics` through other Flutter and Dart Tooling.

Each of the scenarios is for the first time a developer interacts with analytics from devtools.

1. If the developer has interacted with `package:unified_analytics` and is opted in --> `analytics.telemetryEnabled` = `true`
2. If the developer has interacted with `package:unified_analytics` and is opted out --> `analytics.telemetryEnabled` = `false`

3. The developer has never interacted with `package:unified_analytics` --> `analytics.telemetryEnabled` = `true`

For situation 3, we set the `telemetryEnabled` value to `true` the first time the tool is run, but this still doesn't allow events to be sent during that run.

> will this mess up existing DevTools users who are opted into analytics

So if they were opted into devtools historically, but they opted out of analytics with `package:unified_analytics` in a different tool, this would now disable events from those users. This is referring to scenario 2 above